src/index.ts

```diff
@@ -96,8 +99,16 @@ export class WaxJS {
     ) => void;
     metricURL?: string;
     returnTempAccounts?: boolean;
+    chainName?: string;
   }) {
     this.rpc = new JsonRpc(rpcEndpoint);
+
+    this.chainName = chainName || null;
+    this.chainId = null;
+    if (chainName) {
+      this.setChainName(chainName);
```

Comment:
there should be an await here

Reply:
Also we will rename this to `setupChain`

Reply:
@porkchop 
i already update naming in this case
but a bit confuse to call await inside of `constructor`
any other way to handle this case?

Reply:
@sonnguyenwax, @porkchop sync we cannot use await in constructor, what if we calling this.signingApi inside then to make sure this.rpc always initialized beore retrieval?:
```js
this.setupChain(chainName).then(() => {
 this.signingApi = new WaxSigningApi(...)
})
```